src/Components/LanguageSelector/LanguageSelector.jsx

```diff
@@ -21,13 +21,17 @@ const LanguageSelector = () => {
           [languages.find(({ code }) => code === language)].map(
             ({ code, name }) => ({
               value: code,
-              label: name,
+              label: (
+                <SelectOptionWithFlag name={name} foreignLanguageCode={code} />
+              ),
             })
           )[0]
         }
         options={languages.map(({ code, name }) => ({
```

Comment:
maybe we can move this map function to a named function and reuse it.

Reply:
I am not sure if i understand you. Could you give an example for what you mean?